### DISCUSSION TITLE
quarto render will not "finish"

### DISCUSSION THREAD

**Discussion by nucleic-acid** [quarto-dev/quarto-cli#2471] [Q&A]
Since the update to v1.1.189 the process of rendering a website hangs after the last *.qmd file, in my case usually posts.qmd.
The preview will not open and a process called `deno` uses a lot of CPU ressources untill killed.

Also rendering single qmd files will not finish and open the preview.

I'm on Kubuntu 20.04. This problem occurs in RStudio and vscode.
An update to v1.1.251 today did not resolve the issue.

Is this a known issue and is there a way to solve this?
Thanks in advance.

**Comment by mcanouil** [comment]
Could you provide a reproducible example?

**Comment by nucleic-acid** [comment]
Here's an update on this:

When creating a reproducible example from a blank quarto blog, I could not replicate the issue. After some poking around, I found the issue in a blog post where I use both R and Python. Digging further, I realized, that the R chunks are not being evaluated, only the Python chunks.

I can render the post without any problem, but as soon as quarto tries to update the `posts.qmd` it hangs.
When I exclude this post from the posts listing (by setting `draft: true` in that post's YAML) everything builds and previews just fine.

What I don't understand is, why the post content hinders the rendering of the posts listing?

**Comment by nucleic-acid** [comment]
And another update: the problem stopped, as soon as I added `engine: knitr` to the post's YAML and used the R package `{reticulate}` to evaluate the Python code chunks.

Somehow I thought, quarto automatically uses the correct engine for each code cell, but apparently `{reticulate}` is still needed, when R and Python are used?

**Reply by mcanouil** [reply]
Quarto uses the correct engine, but when there are multiple languages, what is the correct?
`reticulate` is not a language.
Since languages can't be completely mixed up, the user has to make a choice about the main engine.

It seems making a reproducible example kind of solved your issue 😏.

**Reply by nucleic-acid** [reply]
Yes it did and helped narrow the problem down. However, unfortunately I'm back at the start again.

After the above changes I could render the post + render the whole blog including all listings without any problem _once_.
After that, all attempts to render the site again or the post, again hangs at the `posts.qmd`.

I will report back, as soon as I have a reproducible example. Thanks for your quick replys and support! Quarto is an awesome software.
